src/com/mendix/widget/GoogleMaps/components/__tests__/Map.spec.ts

```diff
@@ -0,0 +1,201 @@
+            const output = renderMap({ address });
+            const mapComponent = output.instance() as Map;
+            output.find(GoogleMap).prop("onGoogleApiLoaded")();
+
+            expect(mapComponent.state.location.lat).toBe(successMockLocation.lat);
+            expect(mapComponent.state.location.lng).toBe(successMockLocation.lng);
+        });
+
+        it("should display the first marker if multiple locations are found", () => {
+            const output = renderMap({ address: "multipleAddress" });
+            output.find(GoogleMap).prop("onGoogleApiLoaded")();
+
+            const marker = output.find(Marker);
+            expect(marker.prop("lat")).toBe(multipleAddressMockLocation.lat);
+            expect(marker.prop("lng")).toBe(multipleAddressMockLocation.lng);
+        });
+    });
+
+    describe("with an invalid address", () => {
+        it("should not render a marker", () => {
+            const output = renderMap({ address: invalidAddress });
+            output.find(GoogleMap).prop("onGoogleApiLoaded")();
+
+            const marker = output.find(Marker);
+            expect(marker.length).toBe(0);
+        });
+
+        it("should center to the default address", () => {
+            const output = renderMap({ address: invalidAddress });
+            output.find(GoogleMap).prop("onGoogleApiLoaded")();
+
+            const marker = output.childAt(0);
+            expect(marker.prop("center").lat).toBe(defaultCenterLocation.lat);
+            expect(marker.prop("center").lng).toBe(defaultCenterLocation.lng);
+        });
+
+        it("should display an error", () => {
+            spyOn(window.mx.ui, "error").and.callThrough();
+
+            const output = renderMap({ address: invalidAddress });
+            output.find(GoogleMap).prop("onGoogleApiLoaded")();
+
+            expect(window.mx.ui.error).toHaveBeenCalled();
+        });
+    });
+
+    describe("on loading", () => {
+        it("should load the google maps script without API key", () => {
+            const googleMap = renderMap({ address, apiKey: undefined }).childAt(0);
+
+            expect(googleMap.prop("bootstrapURLKeys").key).not.toContain(APIKey);
+        });
+
+        it("should load the google maps script with API key", () => {
+            const googleMap = renderMap({ address, apiKey: APIKey }).childAt(0);
+
+            expect(googleMap.prop("bootstrapURLKeys").key).toContain(APIKey);
+        });
+    });
+
+    describe("with updated the address", () => {
+        it("should change marker location to new address", () => {
+            const output = renderMap({ address });
+            output.find(GoogleMap).prop("onGoogleApiLoaded")();
+
+            const marker = output.find(Marker).at(0);
+            expect(marker.prop("lat")).toBe(successMockLocation.lat);
+            expect(marker.prop("lng")).toBe(successMockLocation.lng);
+
+            const map = output.instance() as Map;
```

Comment:
There is a `setProps` API on shallow renderer that is preferable iso calling lifecycle methods yourself